walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -41,45 +40,40 @@
       return Arrays.asList();
     }
 
-    List<TimeRange> badTimes = getBadTimes(events, attendeesRequested);
-
-    // list to hold all the available times (everything not in badTimes)
-    List<TimeRange> goodTimes = getGoodTimes(badTimes, meetingDuration);
-    return goodTimes;
+    int meetingDuration = (int) request.getDuration();
+    return getGoodTimes(getBusyTimes(events, attendeesRequested), meetingDuration);
```

Comment:
Sorry for the confusion, my previous suggestion was to call getBusyTimes in the getGoodTimes function itself:
```
    List<TimeRange> getGoodTimes(Collection<Event> events, MeetingRequest request) {
        List<TimeRange> busyTimes = getBusyTimes(events, request.getAttendees());
        ....
        ....
   }

```

Alternatively, your previous structure looks good too.
```
   List<TimeRange> busyTimes = getBusyTimes(events, attendeesRequested);
   return getGoodTimes(busyTimes, meetingDuration);
```

Reply:
Oh, this is much better - thanks!